### PR TITLE
Added callback_file & callback_name to default_args DAG level and tests

### PR DESCRIPTION
**Pull request by subbota19**
Hi astro team!

This PR brings the following enhancements:

1) Unittest for DAG-Level Callback using parameters:

**_on_success_callback_name_** & **_on_success_callback_file_**
**_on_failure_callback_name_** & **_on_failure_callback_file_**

[Astro instruction on how to apply these parameters](https://www.astronomer.io/docs/learn/dag-factory#step-4-optional-add-a-dag-level-callback)

2) While I find the existing callback feature useful which allows specifying callback code at any location, this PR takes it a step further by enabling callbacks to be specified within the DAG's _default_args_. With this enhancement, the callbacks will automatically propagate to the task level as well. This update aligns with how [default_args](https://airflow.apache.org/docs/apache-airflow/2.9.2/core-concepts/dags.html#default-arguments) are passed in Airflow. Additionally, I’ve included unit tests.

### CODE REVIEW

**Review by tatiana — COMMENTED** [review]
This is super helpful, @subbota19 , thank you very much for working on this!
I'm happy for us to merge this change once we confirm the tests are passing.

**Comment by subbota19** [comment]
Hi @tatiana !
I've added another commit to address the failed fmt-check step. The code is now formatted correctly, could we please rerun the workflow one more time?

**Review by pankajkoti — APPROVED** [review]
LGTM. Thanks for the contribution @subbota19 

@tatiana can we please check why the project settings is not asking to rebase/update the PR from main before merging?

**Comment by pankajastro** [comment]
For some reason, I still can't see the update button.  I'm looking for something like below (Cosmos).


Screenshot 2024-10-15 at 1 08 57 AM

**Comment by pankajkoti** [comment]
I was able to test the support for included callbacks in default_args with this change. The CI is happy except for Static-Check failures; since I cannot push to this PR, I will open a follow-up PR to fix the failure. Am going ahead with merging this PR.

**Pull request merged**
